fcrepo-clients/src/main/java/edu/unc/lib/dl/fcrepo4/FileObject.java

```diff
@@ -95,6 +95,31 @@ public BinaryObject addOriginalFile(InputStream contentStream, String filename,
                 filename, mimetype, sha1Checksum, md5Checksum, fileModel);
     }
 
+    /**
+     * Replaces the original file for this file object
+     *
+     * @param contentStream
+     * @param filename
+     * @param mimetype
+     * @param sha1Checksum
+     * @return
+     */
+    public BinaryObject replaceOriginalFile(InputStream contentStream, String filename,
+            String mimetype, String sha1Checksum, String md5Checksum) {
+
+        // Construct the path to where the original file will be created
+        String objectPath = constructOriginalFilePath();
+
+        // Add the OriginalFile use type
+        Model fileModel = ModelFactory.createDefaultModel();
```

Comment:
After thinking about it some, I don't think replaceOriginalFile needs to be providing a model. Nothing is changing about its properties, so this is pretty much refreshing them. If we need to replace its properties at a later point there are other ways to do that.

Reply:
well, that certainly simplifies the method :-)